Added clean_plot() function to capitalize axis titles and remove whitespace

reillyowencooper opened the pull request:
This PR adds new functionality with the `clean_plot` function:

* The function is written as an S3 `ggplot` extension that can be added to the end of a plot call (i.e., `ggplot(mtcars, aes(hp, mpg)) + geom_point() + clean_plot()`
    * The function capitalizes axis titles to conform with the style guide.
    * The function checks for the ggplot geoms, and removes whitespace between the axes and the data if the plot is a heatmap, barplot, density plot, or histogram.
    * The function checks if the geom is a heatmap and removes the axis lines and ticks if so.

This should address #19 and the reference to whitespace in #11

elizabethmcd commented on R/clean_plot.R:
```diff
@@ -0,0 +1,81 @@
+  # Turn axis titles into sentence case
+  if (!is.null(plot$labels$x)) {
+    x_label <- plot$labels$x
+    plot <- plot + ggplot2::labs(x = snakecase::to_sentence_case(x_label, sep_out = " "))
```
I like this, I wasn't aware of the snakecase package!

elizabethmcd commented on NAMESPACE:
```diff
@@ -1,7 +1,9 @@
 # Generated by roxygen2: do not edit by hand
 
+S3method(ggplot_add,clean_plot)
```
Ah I didn't realize that's what this is called

elizabethmcd reviewed (changes requested):
I really like this! 
Can you add to the documentation how to use this method on a plot? If you modify the README.Rmd and knit in RStudio it will auto-generate the main README markdown that is displayed. I think this is enough added to include in the main documentation. 
If you add that and either @mezarque or @taylorreiter take a look while I'm gone and approve feel free to merge in!

reillyowencooper commented:
> I really like this! Can you add to the documentation how to use this method on a plot? If you modify the README.Rmd and knit in RStudio it will auto-generate the main README markdown that is displayed. I think this is enough added to include in the main documentation. If you add that and either @mezarque or @taylorreiter take a look while I'm gone and approve feel free to merge in!

Updated the README! Also fixed the code for the heatmap to work for discrete axes.

elizabethmcd reviewed (approved):
I this is great, thanks for adding/modifying the documentation! Go ahead and merge and I can merge this into my PR, which probably won't be fixed until I'm back because there are larger issues with the bicolor gradients.

taylorreiter commented on R/clean_plot.R:
```diff
@@ -0,0 +1,83 @@
+#' @examples
+#' \dontrun{
+#' library(ggplot2)
+#' library(snakecase)
```
I think you shouldn't need to run `library(snakecase)` if it's used by the function `clean_plot()`. Loading arcadiathemeR should take care of this I think

taylorreiter commented on NAMESPACE:
```diff
@@ -11,5 +13,7 @@ export(scale_fill_arcadia)
 export(show_arcadia_gradients)
 export(show_arcadia_palettes)
 export(theme_arcadia)
+import(ggplot2)
+import(snakecase)
```
Generalized imports in the namespace should be avoided as much as possible. Are both of these necessary to achieve the functionality you were after @reillyowencooper?

Here's documentation for how to avoid these imports: https://r-pkgs.org/dependencies-mindset-background.html#sec-dependencies-namespace

taylorreiter replied:
(because you're creating an s3, you might need these, but if it all possible it would be good to remove them)

reillyowencooper replied:
You're correct - I _think_ I need to keep the ggplot import because of the S3 extension, but definitely can take out the snakecase import. Bad habit from prior R package making. Will change

reillyowencooper replied:
Ok, tracked down what I needed to do - I don't need to import all of ggplot, but I do need to import the `ggplot_add` function. That function then makes `clean_plot` into the S3 object automatically. This means that both `S3method(ggplot_add,clean_plot)` and `importFrom(ggplot,ggplot_add)` go into `NAMESPACE` but I was able to remove `import(ggplot)`.

taylorreiter replied:
ok perfect!

taylorreiter commented on R/clean_plot.R:
```diff
@@ -0,0 +1,83 @@
+  if (contains_tile) {
+    plot <- plot +
+      ggplot2::theme(
+        axis.line = element_blank(),
+        axis.ticks.x = element_blank(),
+        axis.ticks.y = element_blank()
```
the `element_blank()` functions should all be `ggplot2::element_blank()`

taylorreiter reviewed (commented):
@reillyowencooper some non-urgent comments on package import and handling
